add support of rate limit and timeout (#1347)
apollographql/router: docs/source/configuration/traffic-shaping.mdx

```yaml title="router.yaml"
traffic_shaping:
  variables_deduplication: true # Enable the variable deduplication optimization.

>   router: # Rules applied to the router requests

Reviewer: ```suggestion
  router: # Rules applied to requests from clients to the router
```

>     rate_limit: # Only accept 10 requests per 5 secs maximum. If it reaches the limit, requests are waiting

Reviewer: ```suggestion
    rate_limit: # Accept a maximum of 10 requests per 5 secs from a given client. Excess requests must wait.
```

      num: 10
      per: 5sec

>     timeout: 50sec # If request to the router hangs for more than 50secs then cancel the request (by default it's 30 secs)

Reviewer: ```suggestion
    timeout: 50sec # If a request to the router hangs for more than 50 seconds, cancel it. (The default value is `30sec`.)
```

  all:
    query_deduplication: true # Enable query deduplication for all subgraphs.
    compression: br # Enable brotli compression for all subgraphs.

>   subgraphs:

Reviewer: ```suggestion
  subgraphs: # Rules applied to requests from the router to individual subgraphs
```

    products:
      query_deduplication: false # Disable query for the products subgraph.
      compression: gzip # Enable gzip compression only for the products subgraph.

>       rate_limit: # Only accept 10 requests per 5 secs maximum. If it reaches the limit, requests are waiting

Reviewer: ```suggestion
      rate_limit: # Accept a maximum of 10 requests per 5 secs from the router. Excess requests must wait.
```

        num: 10
        per: 5sec

>       timeout: 50sec # If request to subgraph product hangs for more than 50secs then cancel the request (by default it's 30 secs)

Reviewer: ```suggestion
      timeout: 50sec # If a request to the subgraph hangs for more than 50 seconds, cancel it. (The default value is `30sec`.)
```
